borgmatic/borg/check.py

```diff
@@ -102,11 +102,14 @@ def check_archives(repository, storage_config, consistency_config, local_path='b
             repository,
         ) + _make_check_flags(checks, check_last, prefix) + remote_path_flags + lock_wait_flags + verbosity_flags
 
-        # The check command spews to stdout/stderr even without the verbose flag. Suppress it.
-        stdout = None if verbosity_flags else open(os.devnull, 'w')
-
         logger.debug(' '.join(full_command))
-        subprocess.check_call(full_command, stdout=stdout, stderr=subprocess.STDOUT)
+        output = subprocess.check_output(full_command, stderr=subprocess.STDOUT, universal_newlines=True)
+        logger.debug(output)
+
+        borg_output_logger = logging.getLogger('borg_output')
+        borg_output_logger.info('=== borg check ===')
+        borg_output_logger.info(output)
+        borg_output_logger.info('=== borg check end ===')
```

Comment:
I wanted to make as little change as possible to your current behaviour. 

The way this currently works is: your main logger is called 'borgmatic' now, instead of just '' (root).
I added a second logger (named `borg_output`) that is always on log-level INFO. You're right: it could be DEBUG, it could be WARNING. It doesn't matter - it just has the same level everywhere, because I always want to capture the output to process it later.
To not have duplicate output on STDOUT, I renamed the main program logger to 'borgmatic', so it does not care about what's in 'borg_output'.

If I had only one logger, I would always have to print out the borg-output to STDOUT in order to capture it for my Promtheus file output, too

Reply:
The output to stdout is captured, stderr is printed right away. This way a) correlation between stderr and stdout messages is gone, b) it feels like the borg command has stalled, because there is no output until it has finished.

The behavior should more like the tee command. Output to a file-like object and stdout simultaneously and immediately.